### ISSUE TITLE
Apple ProRes and Avid DNxHR do not produce output files

### ISSUE THREAD

**Issue by GoldJohnKing** [Xaymar/obs-StreamFX#911]
### Operating System

Windows 10 1903 and higher

### OBS Studio Version

28.0.1

### StreamFX Version

0.12.0 Alpha 106

### OBS Studio Log

[2022-08-28 16-54-21.txt](https://github.com/Xaymar/obs-StreamFX/files/9482279/2022-08-28.16-54-21.txt)

### OBS Studio Crash Log

No crash

### Current and Expected Behavior

When using StreamFX's encoder, recording file should appear.

This does not happens with 0.11.1 on obs 27.

### Steps to Reproduce the Problem

1. Select any of those encoders provided by StreamFX as obs's recording encoder. For example, Apple ProRes.
2. Save the settings and then start recording.
3. Resources consumption of obs process raised, which probably means the encoder could be working.
4. Go to the recording file output path, there is no file at all.
5. Click Stop Recording, it wont stop at all.
6. Force exit obs, the recording file still not appear.

### Any additional Information we need to know?

Reproduce video has been sent to the discord, please check this link:

https://discord.com/channels/736503051824922685/736531930186776577/1014571619865342045

**Comment by GhostSenseiFR** [comment]
Exact same probleme here trying to encode with Prores or DNx. Worked perfectly fine on version OBS 27.

**Comment by Xaymar** [comment]
#### Confirmed
- ✅ Avid DNxHR
- ✅ ProRES
- ❌ NVENC
- ❌ AMF

Partial repro, title and description are misleading.

**Comment by Xaymar** [comment]
From testing, this appears to be a bug upstream. Debugging reveals that the encoder is correctly encoding, and streaming the raw encoded packets to disk results in a playable file. That means that the underlying problem causing this new issue isn't from StreamFX, but from changes in OBS Studio.

**Comment by Xaymar** [comment]
Here is an excerpt of the log, starting at the first attempt at submitting frames:

```
[StreamFX] send_frame res=0 w=2560 h=1440 pts=0 dts=9223372036854775808 kf=true
[StreamFX] receive_packet res=-11 sz=0 pts=0 dts=0 kf=false
[StreamFX] send_frame res=0 w=2560 h=1440 pts=1 dts=9223372036854775808 kf=true
[StreamFX] receive_packet res=-11 sz=0 pts=0 dts=0 kf=false
[StreamFX] send_frame res=0 w=2560 h=1440 pts=2 dts=9223372036854775808 kf=true
[StreamFX] receive_packet res=-11 sz=0 pts=0 dts=0 kf=false
[StreamFX] send_frame res=0 w=2560 h=1440 pts=3 dts=9223372036854775808 kf=true
[StreamFX] receive_packet res=0 sz=455215 pts=0 dts=0 kf=true
[StreamFX] send_frame res=0 w=2560 h=1440 pts=4 dts=9223372036854775808 kf=true
[StreamFX] receive_packet res=0 sz=455279 pts=1 dts=1 kf=true
[StreamFX] send_frame res=0 w=2560 h=1440 pts=5 dts=9223372036854775808 kf=true
[StreamFX] receive_packet res=0 sz=455418 pts=2 dts=2 kf=true
```

As you can see, as soon as frame 4 is submitted, the first frame is returned as expected and packets are exported. The `received_packet` value is kept at false until that very moment.

**Comment by Xaymar** [comment]
Scratch that, for some reason received_packet is true despite it supposed to be false.

**Comment by Xaymar** [comment]
This exclusively affects Apple ProRes (prores_aw) and Avid DNxHR only. I do not know why, but it appears these codecs freeze muxer, as it does not even print anything to the log.

**Comment by Xaymar** [comment]
I've tried various hack-y solutions to the problem, without any effect so far. As far as the code goes, the subprocess `obs-ffmpeg-mux` is stuck on [this line](https://github.com/obsproject/obs-studio/blob/master/plugins/obs-ffmpeg/ffmpeg-mux/ffmpeg-mux.c#L647) and no longer responds - a similar issue as shown in [#2855](https://github.com/obsproject/obs-studio/issues/2855). This failure case only occurs if the output muxing capture did not send any headers.

**Comment by Xaymar** [comment]
Seems like this is due to the ProRes and Avid DNxHR encoders not providing any extra data at all. It may be necessary to simply return an empty buffer after X frames have passed, as I do not see any other solutions for now.

**Comment by Xaymar** [comment]
Test builds are available in PR #919.

**Comment by GoldJohnKing** [comment]
> Test builds are available in PR #919.

Issue fixed confirmed, thanks for your efforts!

**Comment by GhostSenseiFR** [comment]
> > Test builds are available in PR #919.
> 
> Issue fixed confirmed, thanks for your efforts!

Hey @GoldJohnKing 
It looks like it is working for you right?
I don't understand much of all of this but its still not work for me.
Do I have to do something in order to fix the issue in my OBS ?

Thank you and have a nice day!

**Comment by GoldJohnKing** [comment]
> > > Test builds are available in PR #919.
> > 
> > Issue fixed confirmed, thanks for your efforts!
> 
> Hey @GoldJohnKing 
> It looks like it is working for you right?
> I don't understand much of all of this but its still not work for me.
> Do I have to do something in order to fix the issue in my OBS ?
> 
> Thank you and have a nice day!

Just download the installer from the Github Action of the above PR, install and overwrite the older verson, start OBS, all things goes well then.

**Comment by GhostSenseiFR** [comment]
> > > > Test builds are available in PR #919.
> > > 
> > > 
> > > Issue fixed confirmed, thanks for your efforts!
> > 
> > 
> > Hey @GoldJohnKing
> > It looks like it is working for you right?
> > I don't understand much of all of this but its still not work for me.
> > Do I have to do something in order to fix the issue in my OBS ?
> > Thank you and have a nice day!
> 
> Just download the installer from the Github Action of the above PR, install and overwrite the older verson, start OBS, all things goes well then.

Thank you very much!
